test/extensions/filters/http/ext_proc/ext_proc_grpc_fuzz.cc

```diff
@@ -0,0 +1,315 @@
+      // Create a cluster for our gRPC server pointing to the address that is running the gRPC
+      // server.
+      auto* processor_cluster = bootstrap.mutable_static_resources()->add_clusters();
+      processor_cluster->set_name("ext_proc_server");
+      processor_cluster->mutable_load_assignment()->set_cluster_name("ext_proc_server");
+      auto* address = processor_cluster->mutable_load_assignment()
+                          ->add_endpoints()
+                          ->add_lb_endpoints()
+                          ->mutable_endpoint()
+                          ->mutable_address()
+                          ->mutable_socket_address();
+      address->set_address(Network::Test::getLoopbackAddressString(ipVersion()));
+      address->set_port_value(test_processor_.port());
+
+      // Ensure "HTTP2 with no prior knowledge." Necessary for gRPC.
+      ConfigHelper::setHttp2(
+          *(bootstrap.mutable_static_resources()->mutable_clusters()->Mutable(0)));
+      ConfigHelper::setHttp2(*processor_cluster);
+
+      // Make sure both flavors of gRPC client use the right address.
+      if (ipVersion() == Network::Address::IpVersion::v4) {
+        const auto addr = std::make_shared<Network::Address::Ipv4Instance>(
+            Network::Test::getLoopbackAddressString(ipVersion()), test_processor_.port());
+        setGrpcService(*proto_config_.mutable_grpc_service(), "ext_proc_server", addr);
+      } else {
+        const auto addr = std::make_shared<Network::Address::Ipv6Instance>(
+            Network::Test::getLoopbackAddressString(ipVersion()), test_processor_.port());
+        setGrpcService(*proto_config_.mutable_grpc_service(), "ext_proc_server", addr);
+      }
+
+      // Merge the filter.
+      envoy::config::listener::v3::Filter ext_proc_filter;
+      ext_proc_filter.set_name("envoy.filters.http.ext_proc");
+      ext_proc_filter.mutable_typed_config()->PackFrom(proto_config_);
+      config_helper_.addFilter(MessageUtil::getJsonStringFromMessageOrDie(ext_proc_filter));
+    });
+
+    // Make sure that we have control over when buffers will fill up.
+    config_helper_.setBufferLimits(BufferSize, BufferSize);
+
+    setUpstreamProtocol(Http::CodecType::HTTP2);
+    setDownstreamProtocol(Http::CodecType::HTTP2);
+  }
+
+  IntegrationStreamDecoderPtr sendDownstreamRequest(
+      absl::optional<std::function<void(Http::HeaderMap& headers)>> modify_headers,
+      absl::string_view http_method = "GET") {
+    auto conn = makeClientConnection(lookupPort("http"));
+    codec_client_ = makeHttpConnection(std::move(conn));
+    Http::TestRequestHeaderMapImpl headers{{":method", std::string(http_method)}};
+    if (modify_headers) {
+      (*modify_headers)(headers);
+    }
+    HttpTestUtility::addDefaultHeaders(headers, false);
+    return codec_client_->makeHeaderOnlyRequest(headers);
+  }
+
+  IntegrationStreamDecoderPtr sendDownstreamRequestWithBody(
```

Comment:
Did you consider inheriting from `ExtProcIntegrationTest ` or some common superclass?

Reply:
This could possibly be a lot cleaner, but needs a little bit of exploration on my part. Early on I had trouble inheriting from `ExtProcIntegrationTest` because of the `TestWithParam` class it inherits from. To unblock myself I made a copy of `StreamingIntegrationTest` and inherited from the `BaseGrpcClientIntegrationParamTest`.

This was something I completely forgot to come back to and address more cleanly.

Reply:
I think you need a base class `ExtProcIntegrationBase` and then a `ExtProcIntegrationTest` that extends it for those tests with `TestWithParam`. Then here you extend `ExtProcIntegrationBase`. It's a nice to have, I know you might not have cycles to do it, if you can't, please leave a TODO explaining this cleanup.

Reply:
That sounds like a good approach to me. I added a TODO to top of file with your explanation of the approach